### ISSUE TITLE
Operational Error

### ISSUE THREAD

**Issue by as1403** [baliga-lab/cmonkey2#76]
Hello,

I am having an issue where the cMonkey2 algorithm throws an Operational Error due to the database being locked. This always seems to happen when the program is around ~1400 runs. I have attached a picture for reference. Any assistance would be appreciated and I would be more than happy to provide additional information.

<img width="818" alt="image001" src="https://user-images.githubusercontent.com/8571180/58804670-0a13a300-85d8-11e9-8c27-6ec2a9df2f7f.png">

**Comment by weiju** [comment]
Hi, thanks for reporting this. It looks like you are getting into situation where the sqlite database gets locked. Are you maybe running the cmviewer at the same time ? This is usually the most likely situation that this can happen.

**Comment by as1403** [comment]
Yes, I was running cmviewer at the same time as the program. Is there a way to avoid the sqlite database locking while running the program and cmviewer simultaneously? In the meantime, I will try executing the program without opening cmviewer.

Thanks!
Aadit

**Comment by weiju** [comment]
I see, so the problem is that sqlite databases don't really like access from multiple applications at the same time. It works most of the time, when the calls don't interfere with each other at exactly the same time.
cmonkey2 actually supports different database backends, so it would be possible to use a regular database server such as e.g. MySQL for storing the run. To do so one needs to create an empty database (and probably login credentials, too) in MySQL and provide them to both cmonkey2 and cm2viewer with the --dburl parameter.
